### PR TITLE
Add configurable CORS middleware

### PR DESCRIPTION
**Pull request by weiser**
From an earlier conversation, this PR gives the server the ability to add what custom headers are allowed within CORS preflight requests.

### CODE REVIEW

File: router_test.go [hunk 1]
```diff
@@ -178,6 +179,47 @@ func TestRouterDefault(t *testing.T) {
...
+	cfg.AllowAllOrigins = true
+	cfg.AllowHeaders = append(cfg.AllowHeaders, "Authorization", "X-Istreamplanet-User-Identity")
+
+	r := NewTestRouter(t, CORSHandler(cors.New(cfg)))
+
+	type EchoResponse struct {
+		Value string `json:"value" description:"The echoed back word"`
+	}
+
+	r.Resource("/echo",
```
Comment (weiser) [inline router_test.go:193]: I should just have this handler be a simple "ping" endpoint.

File: router_test.go [hunk 2]
```diff
@@ -178,6 +179,47 @@ func TestRouterDefault(t *testing.T) {
 	_ = NewTestRouter(t)
 }
 
+func TestRouterConfigurableCors(t *testing.T) {
+	cfg := cors.DefaultConfig()
+	cfg.AllowAllOrigins = true
+	cfg.AllowHeaders = append(cfg.AllowHeaders, "Authorization", "X-Istreamplanet-User-Identity")
```
Comment (danielgtaylor-isp) [inline router_test.go:185]: Can we use a different example so we don't leak iSP-specific stuff?

Comment (weiser) [reply router_test.go]: At 3dc764f

**Comment by weiser** [comment]
Setting custom CORS preflight allowable headers: d0df19d

**Comment by weiser** [comment]
> Disabling CORS (I would set a blank handler function that does nothing)

Couldn't that be done by just not providing a cors middleware at all?  like going from:
```
g := gin.New()
g.Use(cors.Default())
r := huma.NewRouter("My API", "1.0.0", huma.WithGin(g))
```

to 
```
g := gin.New()
r := huma.NewRouter("My API", "1.0.0", huma.WithGin(g))
```
?

**Comment by weiser** [comment]
Once yuo're Ok with these changes, I'm going to rebase so the history looses references to isp-specific headers.

**Comment by danielgtaylor** [comment]
Let's skip the disabling CORS docs for now, this looks good, thanks :+1:

File: README.md [hunk 3]
```diff
@@ -615,6 +615,20 @@ g.NoRoute(huma.Handler404())
...
+// CORS: Allow non-standard headers "Authorization" and "X-My-Header" in preflight requests
+cfg := cors.DefaultConfig()
+cfg.AllowAllOrigins = true
+cfg.AllowHeaders = append(cfg.AllowHeaders, "Authorization", "X-My-Header")
+
+// And manual settings:
+g := gin.New()
+g.Use(huma.CORSHandler(cors.New(cfg)))
+g.NoRoute(huma.Handler404())
+r := huma.NewRouter("My API", "1.0.0", huma.WithGin(g))
```
Comment (danielgtaylor) [inline README.md:629]:
```suggestion
// And manual settings:
r := huma.NewRouter("My API", "1.0.0", huma.CORSHandler(cors.New(cfg)))
```

File: README.md [hunk 4]
```diff
@@ -615,6 +615,20 @@ g.NoRoute(huma.Handler404())
 r := huma.NewRouter("My API", "1.0.0", huma.WithGin(g))
 ```
 
+If you would like CORS preflight requests to allow specific headers, do the following:
+```go
```
Comment (danielgtaylor) [inline README.md:619]:
```suggestion

## Custom CORS Handler

If you would like CORS preflight requests to allow specific headers, do the following:

```go
```

**Pull request merged**